src/Logging/Logging.Console/src/ConsoleLogger.cs

```diff
@@ -73,6 +73,31 @@ public virtual void WriteMessage(LogLevel logLevel, string logName, int eventId,
                 logBuilder = new StringBuilder();
             }
 
+            LogMessageEntry entry;
+            if (Options.Format == ConsoleLoggerFormat.Default)
+            {
+                entry = CreateLogMessageDefault(logBuilder, logLevel, logName, eventId, message, exception);
+            }
+            else if (Options.Format == ConsoleLoggerFormat.Systemd)
+            {
+                entry = CreateLogMessageSystemd(logBuilder, logLevel, logName, eventId, message, exception);
+            }
+            else
+            {
+                throw new ArgumentOutOfRangeException(nameof(Options.Format));
```

Comment:
Should this be an `InvalidOperationException` instead?
So that it indicates a wrong option is passed in.

Reply:
I don't think we should throw here. It's way too late. Throw on startup so random log statements don't start throwing.

Reply:
I've moved the check to the Options class. I've kept the exception in ConsoleLogger to get rid of the compiler error that entry is not assigned.

Reply:
This code should be change now to just assert that the right format was set instead of throwing.

Reply:
I'm not sure how you want to express the assert in code?

Reply:
In the beginning of the method.

`Debug.Assert(Options.Format >= Default && Options.Format <= Systemd)`